### PR TITLE
[FLINK-27502] Add file Suffix to data files

### PR DESCRIPTION
**Pull request by hililiwei**
## What is the purpose of the change

Add file Suffix to data files, close [FLINK-27502](https://issues.apache.org/jira/browse/FLINK-27502)

## Verifying this change

org.apache.flink.table.store.format.FileFormatSuffixTest

### CODE REVIEW

File: flink-table-store-format/pom.xml [hunk 1]
```diff
@@ -190,6 +190,13 @@ under the License.
                 </exclusion>
             </exclusions>
         </dependency>
+
+        <dependency>
+            <groupId>junit</groupId>
```
Comment (JingsongLi) [inline flink-table-store-format/pom.xml:195]: Do we need this dependency? I see you are using junit 5.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/format/FileFormat.java [hunk 2]
```diff
@@ -53,10 +53,20 @@
...
+
+    protected FileFormat(String formatIdentifier) {
+        this.formatIdentifier = formatIdentifier;
+    }
+
     protected abstract BulkDecodingFormat<RowData> getDecodingFormat();
 
     protected abstract EncodingFormat<BulkWriter.Factory<RowData>> getEncodingFormat();
 
+    public String getSuffix() {
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/format/FileFormat.java:66]: Can we unify the naming? I see a lot of places where it's `formatIdentifier`, but here it's `getSuffix`.
We can use `getFormatIdentifier` here.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/DataFilePathFactory.java [hunk 3]
```diff
@@ -34,20 +34,30 @@ public class DataFilePathFactory {
     private final String uuid;
 
     private final AtomicInteger pathCount;
+    private final String formatIdentifier;
 
     public DataFilePathFactory(Path root, String partition, int bucket) {
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/DataFilePathFactory.java:39]: I think we can do without this constructor, the `formatIdentifier` is not a nullable field, and the following can be done without having the special treatment of null.

**Review by JingsongLi — COMMENTED** [review]
Thanks @hililiwei for the contribution! Looks nice~ I left some comments.

File: flink-table-store-format/pom.xml (hunk 1 above)
Comment (hililiwei) [reply flink-table-store-format/pom.xml]: deleted

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/DataFilePathFactory.java (hunk 3 above)
Comment (hililiwei) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/DataFilePathFactory.java]: Fair enough, deleted.

**Comment by hililiwei** [comment]
> Thanks @hililiwei for the contribution! Looks nice~ I left some comments.

thank you for review, PTAL.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java [hunk 4]
```diff
@@ -106,7 +114,12 @@ public Path toTmpSnapshotPath(long id) {
     }
 
     public DataFilePathFactory createDataFilePathFactory(BinaryRowData partition, int bucket) {
-        return new DataFilePathFactory(root, getPartitionString(partition), bucket);
+        return createDataFilePathFactory(partition, bucket, formatIdentifier);
+    }
+
+    public DataFilePathFactory createDataFilePathFactory(
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java:120]: Why introduce this method? There is a `formatIdentifier` in here.

**Comment by JingsongLi** [comment]
You can run a  `mvn spotless:apply` after modification.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java (hunk 4 above)
Comment (hililiwei) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java]: Sorry, I submitted the wrong one. Has been updated.

**Comment by hililiwei** [comment]
> You can run a `mvn spotless:apply` after modification.

done.

**Comment by JingsongLi** [comment]
```
Error:  Failures: 
Error:    DataFilePathFactoryTest.testNoPartition:46 
expected: /tmp/junit13044672734532131/bucket-123/data-de33b78b-2802-49a0-8ec7-cbeeed7af049-0
 but was: /tmp/junit13044672734532131/bucket-123/data-de33b78b-2802-49a0-8ec7-cbeeed7af049-0.orc
Error:    DataFilePathFactoryTest.testWithPartition:64 
expected: /tmp/junit3939205851475787579/dt=20211224/bucket-123/data-8a204f50-ad09-4e55-a41f-39c94da0863c-0
 but was: /tmp/junit3939205851475787579/dt=20211224/bucket-123/data-8a204f50-ad09-4e55-a41f-39c94da0863c-0.orc
[INFO] 
```

Test failure, you need to adjust some testing codes too.

**Comment by hililiwei** [comment]
Updated and passed the fork repo CI. Please let it run again, i will see if there are any other mistakes. @JingsongLi

**Review by JingsongLi — APPROVED** [review]
Looks good to me! Thanks for your contribution!

**Comment by hililiwei** [comment]
thanks for review.

**Pull request merged**